media/views.py

```diff
@@ -13,6 +13,30 @@ def get(self, request):
         except Image.DoesNotExist:
             return JsonResponse({'error': 'Image not found'}, status=404)
 
-        response = HttpResponse(image.image_data, content_type='image/png')
-        response['Content-Disposition'] = 'attachment; filename="{}.png"'.format(image.title)
-        return response
+        try:
+            response = HttpResponse(image.image_data, content_type='image/png')
+            response['Content-Disposition'] = 'attachment; filename="{}.png"'.format(image.title)
+            return response
+        except Exception as e:
+            return JsonResponse({'error': str(e)}, status=500)
+
+
+class CSVView(View):
+    def get(self, request):
+        csv_uuid = request.GET.get('uuid')
+        try:
+            csv = CSVFile.objects.get(uuid=csv_uuid)
+        except CSVFile.DoesNotExist:
+            return JsonResponse({'error': 'CSV not found'}, status=404)
+
+        try:
+            csv_text = csv.fetch_csv_from_blob()
+            response = HttpResponse(content_type='text/csv')
+            title = csv.title
+            if not title or title == 'Untitled':
+                title = 'Untitled-{}'.format(csv.uuid)
+            response['Content-Disposition'] = 'attachment; filename="{}.csv"'.format(title)
```

Comment:
title already has .csv appended in it as per line 58 in basic_result_interpreter.py 

this is generating file with names like title.csv.csv

Reply:
Fixed this